### ISSUE TITLE
Unable to use custom Logger due to config-validator

### ISSUE THREAD

**Issue by Vortex375** [deepstreamIO/deepstream.io#1020]
I have a custom Logger class like this:
`class LogAdapter extends DeepstreamPlugin implements DeepstreamLogger { ... }`
and I register it statically with
```
import { registerPlugin } from "@deepstream/server/dist/src/config/config-initialiser";
registerPlugin(LOG_ADAPTER_PLUGIN_NAME, LogAdapter);
```

Now, I want to use it. In `config-initializer.ts` in `handleLogger()` I see:
```
if (config.logger.type === 'default') {
    LoggerClass = StdOutLogger
  } else {
    LoggerClass = resolvePluginClass(config.logger, ...);
}
```
So I have to put something else than "default" for the plugin type in my configuration. Let's try it:
```
const DEEPSTREAM_CONFIG: PartialDeepstreamConfig = {
  logger: {
    type: "custom",
    name: LOG_ADAPTER_PLUGIN_NAME
  },
  ...
}
```

Unfortunately that does not work:
```
Error: Invalid configuration: data.logger.type should be equal to one of the allowed values
    at Object.exports.validate (/home/vortex/workspace/iw/node_modules/@deepstream/server/src/config/config-validator.ts:248:11)
```

config-validator only allows type: "default" for the logger configuration, making it impossible to register a custom log plugin (`handleLogger()` will always use the `StdOutLogger`).

When I add another string as allowed value, it instantly works. However, before I make a pull request, I want to understand the meaning of the "type" field in the plugin configuration and how one is _supposed_ to register a custom logger plugin.

**Comment by yasserf** [comment]
Hello!

Thank you for raising, and yes your right there is an issue where the merge keeps type and hence ignore paths. This has been fixed in the V5 version which should be out very soon.

For now you can do:

deepstream.set(logger, new CustomLogger())

**Comment by yasserf** [comment]
Closing as this is fixed in 5.0
